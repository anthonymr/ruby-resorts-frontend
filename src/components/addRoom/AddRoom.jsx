import React, { useState } from 'react';
import { useDispatch } from 'react-redux';
import {
  Container,
  Grid,
  TextField,
  Button,
  Card,
  CardContent,
  Typography,
  Snackbar,
} from '@mui/material';
import { useAddNewRoomMutation } from '../../services/apiService';
import { addRoomReducer } from '../../redux/mainPage/roomsSlice';

async function fileToBase64(file) {
  return new Promise((resolve, reject) => {
    const reader = new FileReader();
    reader.readAsDataURL(file);
    reader.onload = () => resolve(reader.result);
    reader.onerror = (error) => reject(error);
  });
}

const AddRoom = () => {
  const [roomData, setRoomData] = useState({
    name: '',
    price: '',
    reservationPrice: '',
    reservationFee: '',
    rating: '',
    description: '',
    image: null,
  });

  const [nameError, setNameError] = useState('');
  const [descriptionError, setDescriptionError] = useState('');

  const dispatch = useDispatch();

  const [addNewRoom, { isLoading }] = useAddNewRoomMutation();

  const [successMessage, setSuccessMessage] = useState('');
  const [errorMessage, setErrorMessage] = useState('');

  const handleChange = (e) => {
    const { name, value } = e.target;

    if (['price', 'reservationPrice', 'reservationFee'].includes(name)) {
      const numericValue = Number(value);
      if (numericValue <= 0) {
        return;
      }
    } else if (name === 'rating') {
      const numericValue = Number(value);
      if (numericValue < 0 || numericValue > 5) {
        return;
      }
    }

    setRoomData({ ...roomData, [name]: value });
  };

  const handleImageChange = async (e) => {
    const file = e.target.files[0];
    if (file) {
      try {
        const base64 = await fileToBase64(file);
        setRoomData({ ...roomData, image: base64 });
      } catch (error) {
        console.error('Error converting file to base64:', error);
      }
    }
  };

  const validateForm = () => {
    let valid = true;

    if (roomData.name.length < 3 || roomData.name.length > 55) {
      setNameError('Name should be between 3 and 55 characters');
      valid = false;
    } else {
      setNameError('');
    }

    if (roomData.description.length < 10 || roomData.description.length > 500) {
      setDescriptionError(
        'Description should be between 10 and 500 characters',
      );
      valid = false;
    } else {
      setDescriptionError('');
    }

    return valid;
  };

  const handleSubmit = async (e) => {
    e.preventDefault();

    if (!validateForm()) {
      return;
    }

    const formData = new FormData();
    formData.append('room[name]', roomData.name);
    formData.append('room[full_price]', roomData.price);
    formData.append('room[reservation_price]', roomData.reservationPrice);
    formData.append('room[reservation_fee]', roomData.reservationFee);
    formData.append('room[rating]', roomData.rating);
    formData.append('room[description]', roomData.description);
    formData.append('image', roomData.image);

    try {
      const { data } = await addNewRoom(formData);
      dispatch(addRoomReducer(data));
      setRoomData({
        name: '',
        price: '',
        reservationPrice: '',
        reservationFee: '',
        rating: '',
        description: '',
        image: null,
      });
      setSuccessMessage('Room created successfully.');
    } catch (error) {
      setErrorMessage('Failed to create room. Please try again.');
    }
  };

  const handleSnackbarClose = () => {
    setSuccessMessage('');
    setErrorMessage('');
  };

  return (
    <Container
      sx={{
        maxWidth: '100%',
        minHeight: '100vh',
        margin: '0 auto',
        padding: { xs: '25% 0 0', sm: '2rem' },
        display: 'flex',
        flexDirection: 'column',
        justifyContent: { xs: 'start', sm: 'center' },
        alignItems: 'center',
        position: 'relative',
        overflowX: 'hidden',
      }}
      disableGutters
    >
      <Grid
        container
        spacing={2}
        justifyContent="center"
        alignItems="center"
        style={{ minHeight: '100vh' }}
      >
        <Card style={{ maxWidth: 450, padding: '20px 5px' }}>
          <CardContent>
            <Typography
              variant="h4"
              fontWeight={900}
              color="black"
              letterSpacing="3px"
              sx={{ margin: '2rem' }}
            >
              ADD NEW ROOM
            </Typography>
            <form onSubmit={handleSubmit}>
              <Grid container spacing={2}>
                <Grid item xs={12}>
                  <TextField
                    label="Name"
                    variant="outlined"
                    fullWidth
                    required
                    name="name"
                    value={roomData.name}
                    onChange={handleChange}
                    error={!!nameError}
                    helperText={nameError}
                  />
                </Grid>
                <Grid item xs={12} sm={6}>
                  <TextField
                    type="number"
                    label="Price"
                    variant="outlined"
                    fullWidth
                    required
                    name="price"
                    value={roomData.price}
                    onChange={handleChange}
                  />
                </Grid>
                <Grid item xs={12} sm={6}>
                  <TextField
                    type="number"
                    label="Reservation Price"
                    variant="outlined"
                    fullWidth
                    required
                    name="reservationPrice"
                    value={roomData.reservationPrice}
                    onChange={handleChange}
                  />
                </Grid>
                <Grid item xs={12} sm={6}>
                  <TextField
                    type="number"
                    label="Reservation Fee"
                    variant="outlined"
                    fullWidth
                    required
                    name="reservationFee"
                    value={roomData.reservationFee}
                    onChange={handleChange}
                  />
                </Grid>
                <Grid item xs={12} sm={6}>
                  <TextField
                    type="number"
                    label="Rating"
                    variant="outlined"
                    fullWidth
                    required
                    name="rating"
                    value={roomData.rating}
                    onChange={handleChange}
                  />
                </Grid>
                <Grid item xs={12}>
                  <TextField
                    label="Description"
                    multiline
                    rows={4}
                    variant="outlined"
                    fullWidth
                    required
                    name="description"
                    value={roomData.description}
                    onChange={handleChange}
                    error={!!descriptionError}
                    helperText={descriptionError}
                  />
                </Grid>
                <Grid item xs={12}>
                  <div>
                    <input
                      id="file-upload-input"
                      type="file"
                      name="image"
                      accept="image/*"
                      className="upload-image-input"
                      onChange={handleImageChange}
                    />
                  </div>
                </Grid>
                <Grid item xs={12}>
                  <Button
                    type="submit"
                    variant="contained"
                    color="secondary"
                    fullWidth
                    sx={{
                      borderRadius: '20px',
                      color: 'white',
                      fontSize: '20px',
                    }}
                    disabled={isLoading}
                  >
                    {isLoading ? 'Adding...' : 'Submit'}
                  </Button>
                </Grid>
              </Grid>
            </form>
          </CardContent>
        </Card>
      </Grid>
      <Snackbar
        open={!!successMessage || !!errorMessage}
        autoHideDuration={4000}
        onClose={handleSnackbarClose}
        message={successMessage || errorMessage}
      />
    </Container>
  );
};

export default AddRoom;
