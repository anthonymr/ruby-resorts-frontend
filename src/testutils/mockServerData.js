const userInfoData = {
  id: 1,
  name: 'Test User',
  username: 'testuser',
  email: 'testuser@test.com',
  role: 'admin',
  created_at: '2023-06-17T12:20:06.319Z',
  updated_at: '2023-06-17T12:20:47.606Z',
};

export const roomsData = [
  {
    id: 1,
    name: 'Server Test Suite 1',
    description: 'description of server test suite 1',
    image_url: 'someimageurl1',
  },
  {
    id: 2,
    name: 'Server Test Suite 2',
    description: 'description of server test suite 2',
    image_url: 'someimageurl2',
  },
];

export const hotelsData = [
  {
    id: 1,
    name: 'Test Hotel 1',
    city: 'Test City 1',
    created_at: '2023-06-17T12:15:02.781Z',
    updated_at: '2023-06-17T12:15:02.781Z',
  },
];

export const detailsData = {
  id: 1,
  name: 'Server Test Suite 1',
  description: 'Description of test suite 1',
  full_price: '100.0',
  reservation_price: '10.0',
  reservation_fee: '10.0',
  rating: 5,
  created_at: '2023-06-17T12:15:02.876Z',
  updated_at: '2023-06-17T12:15:02.887Z',
  image_url: 'some image url',
};

export const newRoomData = {
  id: 1,
  name: 'Server New Test Suite',
  description: 'Description of new test suite',
  full_price: '250.0',
  reservation_price: '200.0',
  reservation_fee: '10.0',
  rating: 5,
  created_at: '2023-06-17T12:15:02.876Z',
  updated_at: '2023-06-17T12:15:02.887Z',
  image_url: 'some image url',
};

export const reservationsData = [
  {
    id: 1,
    start_date: '2023-06-18',
    end_date: '2023-06-20',
    user_id: 1,
    hotel_id: 1,
    room_id: 1,
    created_at: '2023-06-18T07:11:15.304Z',
    updated_at: '2023-06-18T07:11:15.304Z',
    amount: '20.0',
    hotel: {
      id: 1,
      name: 'Test Hotel',
      city: 'Test City',
    },
    room: {
      id: 1,
      name: 'Test Suite',
      description:
        'description of test suite 1',
      full_price: '100.0',
      reservation_price: '10.0',
      reservation_fee: '10.0',
      rating: 5,
    },
    user: {
      id: 1,
      name: 'Server Test User 1',
      username: 'testuser',
      email: 'testuser@test.com',
      role: 'admin',
    },
  },
];

export default userInfoData;
