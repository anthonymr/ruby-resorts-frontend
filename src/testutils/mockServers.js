/*eslint-disable*/
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import userInfoData, {
  roomsData,
  hotelsData,
  detailsData,
  reservationsData,
  newRoomData,
} from './mockServerData';

const roomsHandler = [
  rest.get('http://127.0.0.1:3000/api/v1/authentication', (req, res, ctx) => {
    return res(ctx.status(200), ctx.json(userInfoData), ctx.delay(10));
  }),
  rest.get('http://127.0.0.1:3000/api/v1/rooms', (req, res, ctx) => {
    return res(ctx.status(200), ctx.json(roomsData), ctx.delay(10));
  }),
  rest.get('http://127.0.0.1:3000/api/v1/hotels', (req, res, ctx) => {
    return res(ctx.status(200), ctx.json(hotelsData), ctx.delay(1));
  }),
  rest.get('http://127.0.0.1:3000/api/v1/rooms/1', (req, res, ctx) => {
    return res(ctx.status(200), ctx.json(detailsData), ctx.delay(1));
  }),
  rest.get('http://127.0.0.1:3000/api/v1/reservations', (req, res, ctx) => {
    return res(ctx.status(200), ctx.json(reservationsData), ctx.delay(1));
  }),
  rest.delete('http://127.0.0.1:3000/api/v1/rooms/1', (req, res, ctx) => {
    return res(ctx.status(200), ctx.json(detailsData), ctx.delay(1));
  }),
  rest.post('http://127.0.0.1:3000/api/v1/rooms/', (req, res, ctx) => {
    return res(ctx.status(200), ctx.json(newRoomData), ctx.delay(1));
  }),
];

export const server1 = setupServer(...roomsHandler);
