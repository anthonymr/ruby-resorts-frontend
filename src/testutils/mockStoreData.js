import { configureStore } from '@reduxjs/toolkit';
import roomsReducer from '../redux/mainPage/roomsSlice';
import detailsReducer from '../redux/detailsPage/detailsSlice';
import citiesReducer from '../redux/newReservePage/citiesSlice';
import userReducer from '../redux/login/userSlice';
import reservationsReducer from '../redux/myReservationsPage/reservationsSlice';
import { appApi } from '../services/apiService';

const initialsStateData = {
  rooms: {
    rooms: [
      {
        id: 1,
        name: 'Test Suite',
        description: 'Description of Test Suite',
        image_url: 'Some URL',
      },
    ],
    status: 'completed',
  },
  cities: {
    hotels: [
      {
        id: 1,
        name: 'Ruby Test Hotel',
        city: 'Test City',
        created_at: '2023-06-16T03:22:52.703Z',
        updated_at: '2023-06-16T03:22:52.703Z',
      },
    ],
  },
  user: {
    accessToken: null,
    userinfo: {
      id: 1,
      name: 'Admin User',
      username: 'admin',
      email: 'test@test.com',
      role: 'admin',
      created_at: '2023-06-16T03:22:52.683Z',
      updated_at: '2023-06-16T03:22:52.683Z',
    },
    error: '',
    authStatus: 'loggedin',
    dataFetched: true,
  },
  appApi: {
    queries: {},
    mutations: {},
    provided: {},
    subscriptions: {},
    config: {
      online: true,
      focused: true,
      middlewareRegistered: false,
      refetchOnFocus: false,
      refetchOnReconnect: false,
      refetchOnMountOrArgChange: false,
      keepUnusedDataFor: 60,
      reducerPath: 'appApi',
    },
  },
  reservations: {
    reservations: [
      {
        id: 1,
        start_date: '2023-06-16',
        end_date: '2023-06-17',
        user_id: 1,
        hotel_id: 1,
        room_id: 2,
        created_at: '2023-06-16T05:26:03.849Z',
        updated_at: '2023-06-16T05:26:03.849Z',
        amount: '220.0',
        hotel: {
          id: 1,
          name: 'Ruby Amsterdam',
          city: 'Amsterdam',
        },
        room: {
          id: 2,
          name: 'Emerald',
          description:
            'Emerald Suites are second only to our Ruby Suites when it comes to luxury. The suites come with 5-star amenities',
          full_price: '250.0',
          reservation_price: '220.0',
          reservation_fee: '5.0',
          rating: 5,
        },
        user: {
          id: 1,
          name: 'Admin User',
          username: 'admin',
          email: 'test@test.com',
          role: 'admin',
        },
      },
    ],
    status: 'completed',
  },
};

export const setMockStore = (preloadedState) => configureStore({
  reducer: {
    rooms: roomsReducer,
    cities: citiesReducer,
    user: userReducer,
    [appApi.reducerPath]: appApi.reducer,
    details: detailsReducer,
    reservations: reservationsReducer,
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware({
    serializableCheck: false,
  }).concat(appApi.middleware),
  preloadedState,
});

export default initialsStateData;
