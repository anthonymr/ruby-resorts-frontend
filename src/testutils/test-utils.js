import React from 'react';
import PropTypes from 'prop-types';
import { render as rtlRender } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/store';

const render = (ui, options) => {
  const Wrapper = ({ children }) => <Provider store={store}>{children}</Provider>;
  Wrapper.propTypes = {
    children: PropTypes.node.isRequired,
  };

  return rtlRender(ui, { wrapper: Wrapper, ...options });
};

export * from '@testing-library/react';
export { render };
