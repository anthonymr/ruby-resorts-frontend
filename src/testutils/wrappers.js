import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { PropTypes } from 'prop-types';
import { MemoryRouter, Routes, Route } from 'react-router-dom';
import initialStateData, { setMockStore } from './mockStoreData';

export function renderWithProviders(
  ui,
  {
    preloadedState = initialStateData,
    store = setMockStore(preloadedState),
    ...renderOptions
  } = {},
) {
  function Wrapper({ children }) {
    Wrapper.propTypes = {
      children: PropTypes.node.isRequired,
    };
    return <Provider store={store}>{children}</Provider>;
  }

  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}

export const renderWithRouter = (
  ui,
  {
    preloadedState = initialStateData,
    store = setMockStore(preloadedState),
    ...renderOptions
  } = {},
) => {
  function Wrapper({ children }) {
    Wrapper.propTypes = {
      children: PropTypes.node.isRequired,
    };
    return (
      <MemoryRouter initialEntries={['/newreservepage/0']}>
        <Provider store={store}>
          <Routes>
            <Route path="/newreservepage/:roomId" element={children} />
          </Routes>
        </Provider>
      </MemoryRouter>
    );
  }
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
};
