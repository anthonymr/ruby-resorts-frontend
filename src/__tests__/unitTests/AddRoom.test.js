import { render } from '../../testutils/test-utils'; // Import the custom render function
import AddRoom from '../../components/addRoom/AddRoom';

it('renders correctly', () => {
  const { asFragment } = render(<AddRoom />);
  expect(asFragment()).toMatchSnapshot();
});
