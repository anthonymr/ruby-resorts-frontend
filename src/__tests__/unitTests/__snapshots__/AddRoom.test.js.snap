// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg MuiContainer-disableGutters css-ow71lp-MuiContainer-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-6s1xft-MuiGrid-root"
      style="min-height: 100vh;"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
        style="max-width: 450px; padding: 20px 5px;"
      >
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <h4
            class="MuiTypography-root MuiTypography-h4 css-1vc6jzf-MuiTypography-root"
          >
            ADD NEW ROOM
          </h4>
          <form>
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
              >
                <div
                  class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="false"
                    for=":r0:"
                    id=":r0:-label"
                  >
                    Name
                    <span
                      aria-hidden="true"
                      class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                    >
                       *
                    </span>
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                      id=":r0:"
                      name="name"
                      required=""
                      type="text"
                      value=""
                    />
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-yjsfm1"
                      >
                        <span>
                          Name *
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 css-1b3l6lk-MuiGrid-root"
              >
                <div
                  class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="false"
                    for=":r1:"
                    id=":r1:-label"
                  >
                    Price
                    <span
                      aria-hidden="true"
                      class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                    >
                       *
                    </span>
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                      id=":r1:"
                      name="price"
                      required=""
                      type="number"
                      value=""
                    />
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-yjsfm1"
                      >
                        <span>
                          Price *
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 css-1b3l6lk-MuiGrid-root"
              >
                <div
                  class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="false"
                    for=":r2:"
                    id=":r2:-label"
                  >
                    Reservation Price
                    <span
                      aria-hidden="true"
                      class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                    >
                       *
                    </span>
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                      id=":r2:"
                      name="reservationPrice"
                      required=""
                      type="number"
                      value=""
                    />
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-yjsfm1"
                      >
                        <span>
                          Reservation Price *
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 css-1b3l6lk-MuiGrid-root"
              >
                <div
                  class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="false"
                    for=":r3:"
                    id=":r3:-label"
                  >
                    Reservation Fee
                    <span
                      aria-hidden="true"
                      class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                    >
                       *
                    </span>
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                      id=":r3:"
                      name="reservationFee"
                      required=""
                      type="number"
                      value=""
                    />
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-yjsfm1"
                      >
                        <span>
                          Reservation Fee *
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 css-1b3l6lk-MuiGrid-root"
              >
                <div
                  class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="false"
                    for=":r4:"
                    id=":r4:-label"
                  >
                    Rating
                    <span
                      aria-hidden="true"
                      class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                    >
                       *
                    </span>
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                      id=":r4:"
                      name="rating"
                      required=""
                      type="number"
                      value=""
                    />
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-yjsfm1"
                      >
                        <span>
                          Rating *
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
              >
                <div
                  class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="false"
                    for=":r5:"
                    id=":r5:-label"
                  >
                    Description
                    <span
                      aria-hidden="true"
                      class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                    >
                       *
                    </span>
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline css-8ewcdo-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <textarea
                      aria-invalid="false"
                      class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
                      id=":r5:"
                      name="description"
                      required=""
                      rows="4"
                      style="height: 0px; overflow: hidden;"
                    />
                    <textarea
                      aria-hidden="true"
                      class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
                      readonly=""
                      style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding: 0px; width: 100%;"
                      tabindex="-1"
                    />
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-yjsfm1"
                      >
                        <span>
                          Description *
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
              >
                <div>
                  <input
                    accept="image/*"
                    class="upload-image-input"
                    id="file-upload-input"
                    name="image"
                    type="file"
                  />
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth css-1tiv50z-MuiButtonBase-root-MuiButton-root"
                  tabindex="0"
                  type="submit"
                >
                  Submit
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
